editor/plugins/org.fusesource.ide.launcher.ui/src/org/fusesource/ide/launcher/ui/debug/model/CamelModelPresentation.java

```diff
@@ -40,89 +41,111 @@
 		} else if (element instanceof CamelDebuggerVariable) {
-			return Activator.getDefault().getImage(IMG_CAMEL_DEBUGGER);
+			return plugin.getImage(IMG_CAMEL_DEBUGGER);
 		} else if (element instanceof CamelExchangeVariable) {
-			return Activator.getDefault().getImage(IMG_CAMEL_EXCHANGE);
+			return plugin.getImage(IMG_CAMEL_EXCHANGE);
 		} else if (element instanceof CamelMessageVariable) {
-			return Activator.getDefault().getImage(IMG_CAMEL_MESSAGE);
+			return plugin.getImage(IMG_CAMEL_MESSAGE);
 		} else if (element instanceof CamelProcessorVariable) {
-			return Activator.getDefault().getImage(IMG_CAMEL_PROCESSOR);
+			return plugin.getImage(IMG_CAMEL_PROCESSOR);
 		} else if (element instanceof BaseCamelVariable) {
-			return Activator.getDefault().getImage(IMG_CAMEL_VARIABLE);
+			return plugin.getImage(IMG_CAMEL_VARIABLE);
 		} 
 		
 		return null;
 	}
+
+	private Image getImage(Object camelDebugTarget, Activator plugin) {
+		if(((CamelDebugTarget) camelDebugTarget).isDisconnected()){
+			return plugin.getImage(IMG_CAMEL_DISCONNECTED);
+		} else if(((CamelDebugTarget) camelDebugTarget).isSuspended()){
+			return plugin.getImage(IMG_CAMEL_THREAD_PAUSE);
+		} else {
+			return plugin.getImage(IMG_CAMEL_DEBUG_TARGET);		
+		}
+	}
+
+	private Image getImage(CamelThread camelThread, Activator plugin) {
+		if (camelThread.isSuspended()) {
+			return plugin.getImage(IMG_CAMEL_THREAD_PAUSE);
+		} else if (camelThread.isTerminated()) {
+			return plugin.getImage(IMG_CAMEL_DISCONNECTED);
+		} else {
+			return plugin.getImage(IMG_CAMEL_THREAD_RUN);				
+		}
+	}
+
+	private Image getImage(CamelEndpointBreakpoint breakpoint, Activator plugin) {
+		try {
+			if (breakpoint.isEnabled()) {
+				if(breakpoint instanceof CamelConditionalBreakpoint){
+					return plugin.getImage(IMG_CAMEL_CONDITIONAL_BREAKPOINT_ENABLED);
+				} else {
+					return plugin.getImage(IMG_CAMEL_BREAKPOINT_ENABLED);
+				}
+			} else {
+				return plugin.getImage(IMG_CAMEL_BREAKPOINT_DISABLED);
+			}
+		} catch (CoreException ex) {
+			Activator.getLogger().error(ex);
+		}
+		return null;
+	}
 	
-	/*
-	 * (non-Javadoc)
-	 * @see org.eclipse.jface.viewers.LabelProvider#getText(java.lang.Object)
-	 */
 	@Override
 	public String getText(Object element) {
-		// responsible for the text in the breakpoints view
 		if (element instanceof CamelEndpointBreakpoint) {
 			return element.toString();
+		} else if (element instanceof CamelDebugTarget) {
+			return getText((CamelDebugTarget) element);
 		}
 		return null;
 	}
-	
-	/*
-	 * (non-Javadoc)
-	 * @see org.eclipse.debug.ui.IDebugModelPresentation#computeDetail(org.eclipse.debug.core.model.IValue, org.eclipse.debug.ui.IValueDetailListener)
-	 */
+
+	private String getText(CamelDebugTarget camelDebugTarget) {
+		String name = camelDebugTarget.getName();
+		if(camelDebugTarget.isDisconnected()) {
+			return Messages.disconnected+name;
+		} else if(camelDebugTarget.isSuspended()){
+			return Messages.suspended+name;
```

Comment:
I think that the status should be always to the the right of the label and it will allow to stay coherent with Java debugger